query-translator/src/translate.rs

```diff
@@ -117,14 +123,23 @@ pub struct CombinedSelectQuery {
 }
 
 fn cc_to_select_query<T: Into<Option<u64>>>(projection: Projection, cc: ConjoiningClauses, limit: T) -> SelectQuery {
-    SelectQuery {
-        projection: projection,
-        from: FromClause::TableList(TableList(cc.from)),
-        constraints: cc.wheres
-                       .into_iter()
-                       .map(|c| c.to_constraint())
-                       .collect(),
-        limit: limit.into(),
+    if cc.is_known_empty {
+        SelectQuery {
```

Comment:
I'm working on this: it's possible for both SQL queries and Datalog queries to be known-empty, and that can indeed be represented in enums.

Similarly, it's possible for them to have known results — empty is just a special case of this — and it might be useful to flesh this out later. For example, `[:find ?x . :where [?x :db/ident :foo/bar]]` can at algebrizing time be completely computed without touching the database, if we so choose.